modules/internal/ChapelArray.chpl

```diff
@@ -2964,7 +2860,7 @@ module ChapelArray {
   }
 
   proc =(ref a: domain, b: _tuple) {
-    a._value.clearForIteratableAssign();
+    a.clear();
```

Comment:
`clearForIteratableAssign` was simply calling `clear`. And I couldn't think of why we would need a special clear for iteratable assignments. So, I removed `clearForIteratableAssign` completely.